### ISSUE TITLE
Cosmetic issue in FreeBSD Installer setting Computer Name with FQDN

### ISSUE THREAD

**Issue by SanPollo** [helloSystem/hello#324]
**Steps to reproduce**:

1. Run the **FreeBSD Installer** and click **Continue** until you reach the **Create Local Computer Account** page.

2. Fill out the details including the fully qualified domain name in the **Computer Name** field.

3. Tick the box next to **Enable users to log in over the network (ssh)**.

**Actual behaviour**:

Under **Enable users to log in over the network (ssh)** it displays the FQDN with _.local_ appended to it. e.g. _daemon.mydomain.ro.local_ (and a trailing full stop).

**Expected behaviour**:

_.local_ is only appended if the hostname provided by the user does not include the rest of the FQDN (e.g. _daemon.local_)

Also the the trailing full stop should not added to the end of the name as it is a command, not a sentence, and this is confusing.

![fqdn](https://user-images.githubusercontent.com/6193091/149992959-5aea5e35-6a6c-473b-947c-5960e1caf09a.png)

**Comment by probonopd** [comment]
> Also the the trailing full stop should not added to the end of the name as it is a command, not a sentence, and this is confusing.

This is not a punctuation mark but actually part of the `*.local.` domain - it is beneficial to put it there, at least if I am not mistaken:

http://www.dns-sd.org/trailingdotsindomainnames.html

**Comment by SanPollo** [comment]
@probonopd I stand corrected, and have amended the bug report, however it still shouldn't append .local. to a FQDN i.e. it should say _daemon.mydomain.ro._ not _daemon.mydomain.ro.local._

**Comment by probonopd** [comment]
Interesting point. Do people still expose machines directly to the public Internet, isn't that much too dangerous nowadays? Last time I have seen something like this was in the 90s in a university.

**Comment by SanPollo** [comment]
You're absolutely right, however on the corporate networks I have helped to administer (and my home network too as it's full of random servers and other devices!) a machine always has a FQDN even if it isn't accessible outside the network. Admittedly, on corporate Windows networks, the FQDN is not always set to an actual internet domain, but sometimes to a pseudo domain one such as localdomain.myorganisation.corp.

At home, for example, my domain is bromania.eu and this is the search domain that I have set up in DHCP server settings so whenever I ping a machine from a DHCP client it appends the .bromania.eu to the end of it (you can see it in /etc/resolv.conf for FreeBSD clients). Similarly, those machines with a static address also have bromania.eu as part of their FQDN (and in their search domains).

So, while only www.bromania.eu and wiki.bromania.eu are visible to the outside world, other machines like the netbook running helloSystem - daemon.bromania.eu - is only accessible inside. My DNS is also set up to make sure that any requests to www and wiki are sent via the internal IP and not routed via the internet.

This is just an example of how I do things. From a FreeBSD point of view, to the best of my knowledge the FQDN should be specified in the hostname option of _/etc/rc.conf_. Then typing _hostname_ shows the short name, and _hostname -f_ shows the FQDN. The man page is [here](https://www.freebsd.org/cgi/man.cgi?query=hostname&apropos=0&sektion=1&manpath=FreeBSD+13.0-RELEASE&arch=default&format=html).

And this leads me to another question - I noticed that in helloSystem, the _hostname_ option in _/etc/rc.conf_ is set to _"FreeBSD"_ so the real hostname must be set elsewhere. I was just wondering the reason was for doing this?

**Comment by probonopd** [comment]
> You're absolutely right, however on the corporate networks I have helped to administer (and my home network too as it's full of random servers and other devices!) a machine always has a FQDN even if it isn't accessible outside the network.

True, didn't think of this. So we should not append `.local.` in case the hostname contains `.`.
Does this sound about right?

> And this leads me to another question - I noticed that in helloSystem, the hostname option in /etc/rc.conf is set to "FreeBSD" so the real hostname must be set elsewhere. I was just wondering the reason was for doing this?

For the Live system:
https://github.com/helloSystem/ISO/blob/98b56318ce878a80b640db4478ef6fff6a2eb127/overlays/boot/boot/init_script#L191
